site/migrate-mcq-to-qq.md

```diff
@@ -0,0 +1,290 @@
+5. Federation with the old virtual host should be added to any remaining
+   policies, pointing to the federation upstream created earlier:
+   `"federation-upstream-set":"quorum-migration-upstream"`.
+6. If there is no catch-all policy (applying to queues with pattern `.*`), you must create this policy and point it to the federation upstream. This ensures that every queue in the old virtual host will be federated.
+7. Policies that apply federation rules to exchanges must be removed for the period of the migration to avoid duplicate messages.
+
+Now the modified schema can be loaded into the new virtual host from the Management
+UI or by running the following command from the CLI:
+
+```bash
+rabbitadmin import -V NEW_VHOST NEW_VHOST.json
+```
+
+### Point Consumers to use Quorum Queues in the New Virtual Host
+
+Consumers of the migrated queues can now access the new queues by updating the connection parameters to connect to the new virtual host. The federation links start to pull in messages from the original queues.
+
+As with a blue-green cluster, after all consumers are migrated, you might need to also [add shovels to move the backlog of the original queues to the new queues](#shovel-remaining-messages) more efficiently than federation. For more information, refer to [Drain Messages](https://rabbitmq.com/blue-green-upgrade.html#drain-messages).
+
+### Update the Producers to Use the Exchanges in the new Virtual Host
+
+Once the original queues are empty (or nearly empty if you do not require full message ordering), the producers should be stopped and reconfigured to use the new queue declarations and virtual host like the consumers, and restarted. Federated exchanges in the old virtual host should also be stopped and equivalent exchanges should be added in the new virtual host. The original queues can be removed once they are empty and no messages are passing through them.
+
+Under sufficient system, load messages from the old virtual host will
+not be picked up. If message ordering is important, then ordering should
+be completed in these steps: stop producers, shovel remaining messages to the new
+virtual host, and start consumers on the new virtual host.
+
+### <a id="shovel-remaining-messages" class="anchor" href="#shovel-remaining-messages">Shovel Remaining Messages to the New Virtual Host</a> 
+
+For every non-empty queue in the old virtual host, a shovel needs to be configured. For example:
+
+```bash
+rabbitmqctl set_parameter shovel migrate-QUEUE_TO_MIGRATE \
+  '{"src-protocol": "amqp091", "src-uri": "amqp:///OLD_VHOST", "src-queue": "QUEUE_TO_MIGRATE",
+    "dest-protocol": "amqp091", "dest-uri": "amqp:///NEW_VHOST", "dest-queue": "QUEUE_TO_MIGRATE"}'
+```
+
+After the queue is drained, the shovel can be deleted:
+```bash
+rabbitmqctl clear_parameter shovel migrate-QUEUE_TO_MIGRATE
+```
+## <a id="migrate-in-place" class="anchor" href="#migrate-in-place">Migrate in Place</a>
```

Comment:
The steps for the in-place migration are very vague. Is it worth including them?

Reply:
Thanks @MirahImage who can give me these details including the commands to run to complete these steps ? Can someone test it and then provide these details to me?

Reply:
I think it's not possible for us to provide commands for these steps, because they are user-dependent

Reply:
So @MirahImage when you say, what do you mean? 

> steps for the in-place migration are very vague. Is it worth including them

go into more detail for each of these:

Stop the consumers and producers.
Shovel the messages to a new temporary queue.
Delete the old queue.
Create a new quorum queue with the same name as the original queue.
Shovel the contents of the temporary queue to the new quorum queue.
Configure the consumers to use x-queue-type of quorum and they can be started.

?? who can provide more detail here? or should I root around RMQ OSS docs to find that info? 
Thanks

Reply:
ok for now @MirahImage we will leave the steps as is but perhaps we should consider documenting the steps for 1 use case scenario - is there the more popular one that we know of????